ddpui/api/charts_api.py

```
@@ -1165,177 +1127,100 @@ def get_chart_data_by_id(request, chart_id: int, dashboard_filters: Optional[str
+    try:
+        chart_data = ChartData(
+            title=payload.title,
+            description=payload.description,
+            chart_type=payload.chart_type,
+            schema_name=payload.schema_name,
+            table_name=payload.table_name,
+            extra_config=payload.extra_config,
+        )
+        chart = ChartService.create_chart(chart_data, orguser)
+    except ChartValidationError as e:
+        raise HttpError(400, e.message)
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)